Upgrade to new rhetos

ibarban opened the pull request.

Commit:
Changed nuget to PackageReference format.
Changed .NET Framework version to 4.7.2

Commit:
Implemented new configuration usage

bantolov commented on Plugins/AdminSetup/AdminSetup.csproj:
```diff
@@ -228,11 +63,24 @@
+      <Version>2.0.20710</Version>
+    </PackageReference>
+    <PackageReference Include="Rhetos">
+      <Version>3.1.0-dev*</Version>
```
It the star in `3.1.0-dev*` required? Is it standard for references in csproj?

bantolov replied:
We should update Rhetos version to 4.0.0.

sstublic commented on Plugins/InitAspNetDatabase/Program.cs:
```diff
@@ -37,7 +36,10 @@ static int Main(string[] args)
-                Paths.InitializeRhetosServerRootPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\.."));
+                var rootPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..");
+                var configurationProvider = new Rhetos.ConfigurationBuilder().
+                    AddRhetosAppConfiguration(rootPath).Build();
```
Code style: redundant `Rhetos` namespace qualifier, line breaks. Should read something like this:
```
            var configurationProvider = new ConfigurationBuilder()
                .AddRhetosAppConfiguration(rootPath)
                .Build();
```

sstublic commented on Plugins/AdminSetup/Program.cs:
```diff
@@ -51,7 +45,10 @@ static int Main(string[] args)
-                Paths.InitializeRhetosServerRootPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\.."));
+                var configurationProvider = new Rhetos.ConfigurationBuilder().
+                    AddRhetosAppConfiguration(GetRhetosRootFolder()).Build();
+                LegacyUtilities.Initialize(configurationProvider);
```
Intention is not clear to me here. Probably LegacyUtilities are needed even if try..catch fails? 

However, this configuration is redundant and slightly incompatible (missing `AddConfigurationManagerConfiguration`) with the one created at the start of try...catch block. Maybe create container at the top code level to clearly display when it happens as it also initializes LegacyUtilities implicitly. 

If container creation fails itself, I'm not sure if the code is expected to continue at all?

bantolov replied:
This allows call to SetUpAdminAccount to work even if CreateAdminUserAndPermissions failed. It is a strange feature, but we'll leave it for now.
We have modified the code and added comment to make the intention more clear.

bantolov reviewed (changes requested):
Just changing the PR status.

bantolov committed:
Refactoring legacy utilities initialization.

Commit:
Changed Rhetos version to 4.0.0

bantolov committed:
Same configuration for both command-line utilities.
AddConfigurationManagerConfiguration allows users to override Rhetos configuration in the application's config file.
